File: app/src/main/java/com/mahmoud/daggertest/models/custom/ModelThrowable.java
package com.mahmoud.daggertest.models.custom;

public class ModelThrowable {
    private Throwable throwable;
    private boolean displayMessage;
    private boolean displayViewMessage;

    public boolean isDisplayViewMessage() {
        return displayViewMessage;
    }

    public void setDisplayViewMessage(boolean displayViewMessage) {
        this.displayViewMessage = displayViewMessage;
    }

    public ModelThrowable(Throwable throwable, boolean displayMessage, boolean displayViewMessage) {
        this.throwable = throwable;
        this.displayMessage = displayMessage;
        this.displayViewMessage = displayViewMessage;
    }

    public ModelThrowable(String throwable, boolean displayMessage, boolean displayViewMessage) {
        this.throwable = new Throwable(throwable);
        this.displayMessage = displayMessage;
        this.displayViewMessage = displayViewMessage;
    }

    public ModelThrowable(Throwable throwable, boolean displayMessage) {
        this.throwable = throwable;
        this.displayMessage = displayMessage;
        this.displayViewMessage = false;
    }

    public ModelThrowable(String throwable, boolean displayMessage) {
        this.throwable = new Throwable(throwable);
        this.displayMessage = displayMessage;
        this.displayViewMessage = false;
    }



    public ModelThrowable(String throwable) {
        this.throwable = new Throwable(throwable);

    }

    public ModelThrowable(Throwable throwable) {
        this.throwable = throwable;
    }

    public boolean isDisplayMessage() {
        return displayMessage;
    }

    public void setDisplayMessage(boolean displayMessage) {
        this.displayMessage = displayMessage;
    }

    public Throwable getThrowable() {
        return throwable;
    }

    public void setThrowable(Throwable throwable) {
        this.throwable = throwable;
    }


}

File: app/src/main/java/com/mahmoud/daggertest/models/custom/ModelThrowable.kt
package com.mahmoud.daggertest.models.custom

class ModelThrowable {
    var throwable: Throwable
    var isDisplayMessage = false
    var isDisplayViewMessage = false

    constructor(
        throwable: Throwable,
        displayMessage: Boolean,
        displayViewMessage: Boolean
    ) {
        this.throwable = throwable
        isDisplayMessage = displayMessage
        isDisplayViewMessage = displayViewMessage
    }

    constructor(
        throwable: String?,
        displayMessage: Boolean,
        displayViewMessage: Boolean
    ) {
        this.throwable = Throwable(throwable)
        isDisplayMessage = displayMessage
        isDisplayViewMessage = displayViewMessage
    }

    constructor(throwable: Throwable, displayMessage: Boolean) {
        this.throwable = throwable
        isDisplayMessage = displayMessage
        isDisplayViewMessage = false
    }

    constructor(throwable: String?, displayMessage: Boolean) {
        this.throwable = Throwable(throwable)
        isDisplayMessage = displayMessage
        isDisplayViewMessage = false
    }

    constructor(throwable: String?) {
        this.throwable = Throwable(throwable)
    }

    constructor(throwable: Throwable) {
        this.throwable = throwable
    }

}